GetStream/stream-chat-android
# [CAS-1004] Themeing MessageListHeaderView

https://stream-io.atlassian.net/browse/CAS-1004

### 🎯 Goal

Make `MessageListHeaderView` customizable via style transformer (`TransformStyle.messageListHeaderStyleTransformer`) and theme attribute (`StreamUiTheme.streamUiMessageListHeaderStyle`)

### 🛠 Implementation details

Overall, there are 3 ways to customize the look and feel of `MessageListHeaderView`:
 1.  Via `streamUiMessageListHeaderStyle` theme attribute
 2. Via `MessageListHeaderView` layout attributes (overrides pt.1)
 3. Via `TransformStyle.messageListHeaderStyleTransformer` (overrides pt.1 and pt.2)

### 🎨 UI Changes & Testing

1. Try overriding online text color in the theme of our sample app:

```
<style name="StreamTheme" parent="@style/StreamUiTheme">
      <item name="streamUiMessageListHeaderStyle">@style/MessageListHeaderTheme</item>
</style>


<style name="MessageListHeaderTheme" parent="StreamUi.MessageListHeader" >
        <item name="streamUiMessageListHeaderDefaultLabelTextColor">@color/stream_ui_accent_green</item>
</style>
```

<img src="https://user-images.githubusercontent.com/9600921/123812244-9a7a5a00-d8fc-11eb-9fbd-608ab013baef.png" width="40%">

2. Try overriding online text color via layout attributes (this should override corresponding theme attribute):

```
<io.getstream.chat.android.ui.message.list.header.MessageListHeaderView
        android:id="@+id/messagesHeaderView"
        android:layout_width="0dp"
        android:layout_height="wrap_content"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        app:streamUiMessageListHeaderDefaultLabelTextColor="@color/stream_ui_accent_red" />
```        

<img src="https://user-images.githubusercontent.com/9600921/123812758-06f55900-d8fd-11eb-919f-60c6e58b9004.png" width="40%">

### ☑️ Checklist


### 🎉 GIF

![giphy (7)](https://user-images.githubusercontent.com/9600921/123813940-f72a4480-d8fd-11eb-8f17-416f1e1fe5fd.gif)
